### ISSUE TITLE
Как заблокировать загрузку apk - файлов, в том числе по 302-м редиректам ?

### ISSUE THREAD

**Issue by Baltazar500** [AdguardTeam/AdguardFilters#173065]
### Prerequisites

- [X] This site DOES NOT contain sexually explicit material, otherwise use [NSFW-specific form](./new?template=bug_report_NSFW.yml);
- [X] Filters were updated before reproducing an issue;
- [X] AdGuard product version is up-to-date;
- [X] Browser version is up-to-date;
- [X] If the site or app is broken, disabling AdGuard protection resolves an issue.

### What product do you use?

AdGuard for Android

### AdGuard version

3.6

### What type of problem have you encountered?

Missed ads or ad leftovers

### Which browser(s) do you use?

Chrome

### Which device do you use?

Mobile

### Where is the problem encountered?

https://appbazar.am

### Ad Blocking

AdGuard Base filter

### Add your comment and screenshots

1. 

2. Screenshots
<details><summary>Screenshot 1:</summary>

</details>

<details><summary>Screenshot N:</summary>

</details>

### Privacy

- [X] I agree to follow this condition

**Comment by Alex-302** [comment]
`.apk^$important`
Сайты с APK тоже может ломать.

**Comment by Baltazar500** [comment]
Что-то не прокатывает. Всё равно качается. Даже если срезать до

>https://api.appbazar.ru/api/v1/store_versions/latest/build_url

С гитхаба apk-ки тоже apk-ки качаются.

**Comment by zloyden** [comment]
Такое можно попробовать.
```adblock
||api.appbazar.ru/api/v*/store_versions/latest/build_url|$document
```

**Comment by Baltazar500** [comment]
>Такое можно попробовать.

Работает, но это решение лишь для конкретного сайта - примера. Как перекрыть загрузку apk (равно как и любого другого указанного расширения) до сих пор не ясно. Эти костыли не от хорошей жизни нужны, а чтобы дополнительно обезопасить смарты родственников от мошенников :)

**Comment by Alex-302** [comment]
Лучше фича-реквест для антитрекинга откройте в репозитории приложения.

**Comment by Baltazar500** [comment]
>Лучше фича-реквест для антитрекинга откройте в репозитории приложения.

Ну т.е. правилами подобное не решить ? Реализация подобного я так понимаю, повлечёт обновление ядра и собственно приложения ? Я правильно понимаю ?

**Comment by Alex-302** [comment]
Да. Ну в целом функция не лишняя - рекламные сети на всяких пиратских сайтах работают в т.ч. с мошенниками, и есть вероятность случайно установить какой-нибудь руткит.

В винде подобное - часть родительского контроля (не особо полезная реализация, т.к. вредоносы и просто варез распространяется в архивах, иногда с паролем - чтобы антивирусы не сканировали)

<details>
<summary>Details</summary>

![image](https://github.com/AdguardTeam/AdguardFilters/assets/8361299/460ee061-1bf6-4726-bcde-b78ad3ead26d)


</details>

**Comment by Baltazar500** [comment]
>Да. Ну в целом функция не лишняя - рекламные сети на всяких пиратских сайтах работают в т.ч. с мошенниками, и есть вероятность случайно установить какой-нибудь руткит.

ОК. Организую запрос. А в виде userjs как временное решение, можно что-то организовать ? Нагуглилось что-то похожее, но я хз как это допилить для всех запросов с apk-файлами.
```

// ==UserScript==
// @name         YouTube URL Blocker
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  Block GET requests to specific URLs on YouTube.
// @author       me
// @match        https://www.youtube.com/*
// @grant        GM_xmlhttpRequest
// ==/UserScript==

(function() {
    'use strict';

    var blockedURL = "https://suggestqueries-clients6.youtube.com/complete/search";

    document.addEventListener("beforesendheaders", function(e) {
        var requestDetails = e.detail;
        if (requestDetails.method === "GET" && requestDetails.url.startsWith(blockedURL)) {
            e.preventDefault();
            console.log("GET request to blocked URL blocked:", requestDetails.url);
        }
    });

})();

```

**Comment by Baltazar500** [comment]
@Alex-302 , тут https://github.com/AdguardTeam/AdguardForAndroid/issues/5229#issuecomment-1960775099 @sfionov предложил вместо 

>.apk^$important 

использовать 

>.apk^$all

т.к. первое "не работает из-за неприятного бага в CoreLibs" :/ 

>.apk^$all

действительно отрабатывает, как и задумывалось. Правда вышибает на заглушку, где всё равно можно тыкнуть "продолжить", но это уже намного лучше, чем ничего :)
